extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -185,6 +196,28 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
       productForPdp = shoppingDataPerPage[0];
     }
 
+    //pdp view
+    if (productForPdp) {
+      this.variableService_.onVariableUpdate(
+        AnalyticsVariable.STORY_SHOPPING_PRODUCT_ID,
+        productForPdp.productId
+      );
+    }
+    //plp view with more than one product on the page
+    else {
+      const pageId = this.pageEl_.id;
+      this.variableService_.onVariableUpdate(
+        AnalyticsVariable.STORY_PAGE_ID,
```

Comment:
Do we need to set this variable here?
It looks like it's already getting set on page change.
https://github.com/ampproject/amphtml/blob/main/extensions/amp-story/1.0/variable-service.js#L92

Reply:
Yeah, we can remove this line then.